### PR TITLE
EL8 Dockerfiles for rpm package builds

### PR DESCRIPTION
**Pull request by punkrokk**
These are the EL8 Docker containers

Part of effort to add EL8 support.

See https://github.com/StackStorm/st2-packages/pull/627

### CODE REVIEW

File: packagingbuild/centos8/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# St2 package build debs
+RUN yum -y install \
+    openldap-devel
+
+# Enable remote pubkey access
+RUN mkdir /root/.ssh && chmod 700 /root/.ssh && \
+    echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCdCmmPjsOBWRXc+PKdgDRrsciNjp25zTacyz8Gdkln2ma046brOYXAphhp/85DKgHtANBBt3cl4+HnpDbmAfyq2qZT7hWzAbMxtq0Sj+yyFyUdreXoe4gEKyxpV6o8p/R/XzEcawvqX/vFc5EIFmvTdamxZs9DQmOE5AZMzUB18Kerkrb0/arUcZ8iMi9Ng9a18avow+7oUFZ6Oub7ISz/dkIRojaKO/2paJZ4p+v7ZLn7Hq8TUeBkgAlx872oh8J8linhIq17zK6x4MGL8qiurp2hnfe0cuCxwcsYGy+4DfK51+E2vks6FprCIfF5hIdz26euPn67/YpM0F0b5nXF busybee@drone" >> /root/.ssh/authorized_keys
+
+# Create busybee credentials and make busybee pkey available for root
+COPY busybee*  /root/.ssh/
```
Comment (armab) [inline packagingbuild/centos8/Dockerfile]: Looks like the dir is missing this `busybee` SSH private key that needs to copied to the image.

See https://github.com/StackStorm/st2packaging-dockerfiles/pull/69/files for example.

File: packagingtest/centos8/systemd/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# St2 package build debs
+RUN yum -y install \
+    openldap-devel
+
+# Enable remote pubkey access
+RUN mkdir /root/.ssh && chmod 700 /root/.ssh && \
+    echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCdCmmPjsOBWRXc+PKdgDRrsciNjp25zTacyz8Gdkln2ma046brOYXAphhp/85DKgHtANBBt3cl4+HnpDbmAfyq2qZT7hWzAbMxtq0Sj+yyFyUdreXoe4gEKyxpV6o8p/R/XzEcawvqX/vFc5EIFmvTdamxZs9DQmOE5AZMzUB18Kerkrb0/arUcZ8iMi9Ng9a18avow+7oUFZ6Oub7ISz/dkIRojaKO/2paJZ4p+v7ZLn7Hq8TUeBkgAlx872oh8J8linhIq17zK6x4MGL8qiurp2hnfe0cuCxwcsYGy+4DfK51+E2vks6FprCIfF5hIdz26euPn67/YpM0F0b5nXF busybee@drone" >> /root/.ssh/authorized_keys
+
+# Create busybee credentials and make busybee pkey available for root
+COPY busybee*  /root/.ssh/
```
Comment (armab) [inline packagingtest/centos8/systemd/Dockerfile]: Same here with SSH key

**Comment by punkrokk** [comment]
Ok - what does that key do? 

Cheers,

JP

File: packagingbuild/centos8/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#
+
+# Install development tools and python for EL8
+RUN yum -y install python3 python2 python3-devel rpmdevtools \
+&& sudo alternatives --set python /usr/bin/python2
+
+# Install fresh pip and co
+RUN curl https://bootstrap.pypa.io/get-pip.py | \
+    python3 - \
+    virtualenv==16.6.0 pip==19.3.1 wheel setuptools; \
```
Comment (armab) [inline packagingbuild/centos8/Dockerfile]: Please use `pip==19.1.1` to keep the versioning consistent across the all OS builds envs.

https://github.com/StackStorm/st2packaging-dockerfiles/commit/de3f543bf62e90d7cf11c5377939251119137c3a

File: packagingtest/centos7/systemd/Dockerfile [hunk 4]
```diff
@@ -85,6 +80,7 @@ RUN yum -y update; \
     yum -y install systemd; yum clean all;
 
 RUN cd /lib/systemd/system/sysinit.target.wants/; ls -1 | grep -v systemd-tmpfiles-setup.service | xargs rm; \
+    rm -f /lib/systemd/system/multi-user.target.wants/*;\
```
Comment (armab) [inline packagingtest/centos7/systemd/Dockerfile]: Please revert this line change.
See https://github.com/StackStorm/st2packaging-dockerfiles/pull/43/ for context.

Comment (armab) [reply packagingtest/centos7/systemd/Dockerfile]:
```suggestion
```

File: packagingbuild/centos8/Dockerfile (hunk 1 above)
Comment (armab) [reply packagingbuild/centos8/Dockerfile]: The key is used to SSH between the containers for example when one container trigers the build inside another container, transfer artifacts, etc.

Comment (punkrokk) [reply packagingbuild/centos8/Dockerfile]: Gotcha. Resolved

File: packagingtest/centos8/systemd/Dockerfile (hunk 2 above)
Comment (punkrokk) [reply packagingtest/centos8/systemd/Dockerfile]: fixed

File: packagingbuild/centos8/Dockerfile (hunk 3 above)
Comment (punkrokk) [reply packagingbuild/centos8/Dockerfile]: done

File: packagingtest/centos7/systemd/Dockerfile (hunk 4 above)
Comment (punkrokk) [reply packagingtest/centos7/systemd/Dockerfile]: FIxed. Also fixed in https://github.com/StackStorm/st2packaging-dockerfiles/blob/master/packagingbuild/Dockerfile.template

File: packagingbuild/Dockerfile.template [hunk 5]
```diff
@@ -13,12 +13,24 @@ RUN yum -y install \
...
     mercurial \
     openssh \
     subversion
 
+    {% if (version == '8') %}
+RUN yum install -y yum-utils \
+    && dnf config-manager --enable PowerTools \
+    && yum install -y epel-release \
+    && yum -y install \
+    ImageMagick \
```
Comment (nmaludy) [inline packagingbuild/Dockerfile.template]: What is ImageMagick needed for?

**Comment by punkrokk** [comment]
Was already there. I just moved it down because EL8

Cheers,

JP

**Comment by armab** [comment]
After looking at build failures, this is still an issue:
https://github.com/StackStorm/st2packaging-dockerfiles/pull/74#pullrequestreview-333634767
`buildpack/centos8/Dockerfile` is absent, - we'll need it.

**Comment by armab** [comment]
>  ci/dockercloud (/packagingbuild/trusty) — Your tests failed in Docker Cloud 

As for `trusty` Dockerfile build failing, - this could be ignored for now as we'll deprecate it per Roadmap: https://docs.stackstorm.com/roadmap.html#id1

File: packagingtest/centos7/systemd/Dockerfile [hunk 6]
```diff
@@ -72,19 +67,14 @@ RUN sed -ri 's/^session\s+required\s+pam_loginuid.so$/session optional pam_login
...
-
 RUN yum -y install nc net-tools
 
 ENV container docker
 
 RUN yum -y update; \
     yum -y install systemd; yum clean all;
 
-RUN cd /lib/systemd/system/sysinit.target.wants/; ls -1 | grep -v systemd-tmpfiles-setup.service | xargs rm; \
+RUN cd /lib/systemd/system/sysinit.target.wants/; ls -1 | grep -v systemd-tmpfiles-setup.service | xargs rm; \rm -f /lib/systemd/system/multi-user.target.wants/*; \
```
Comment (armab) [inline packagingtest/centos7/systemd/Dockerfile]: Per previous comment (https://github.com/StackStorm/st2packaging-dockerfiles/pull/74#discussion_r359076082), please don't change this line.

```suggestion
RUN cd /lib/systemd/system/sysinit.target.wants/; ls -1 | grep -v systemd-tmpfiles-setup.service | xargs rm; \
```

Comment (punkrokk) [reply packagingtest/centos7/systemd/Dockerfile]: done

Comment (armab) [reply packagingtest/centos7/systemd/Dockerfile]: This still needs to be addressed.
We don't need `rm -f /lib/systemd/system/multi-user.target.wants/*;` line in CentOS Dockerfiles

File: packagingbuild/centos8/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+#
+# Buildenv is special environment for generating debian packages. It provides:
+#   - All needed pre-installed development packages
+#   - SSH access for build executor.
+#
+
+# Install development tools and python for EL8
+RUN yum -y install python3 python2 python3-devel rpmdevtools
+&& sudo alternatives --set python /usr/bin/python2
```
Comment (armab) [inline packagingbuild/centos8/Dockerfile]: Docker build failure for this line: 

```
Dockerfile parse error line 89: unknown instruction: &&
```

Comment (punkrokk) [reply packagingbuild/centos8/Dockerfile]: fixed

**Comment by punkrokk** [comment]
StackStorm/st2#4761

**Pull request merged**